### PR TITLE
Extend SimpleCache to allow for unbounded growth

### PR DESCRIPTION
**Pull request by aaronwinter**
In this PR, we allow users to create a (SimpleCache) cache with no artificial upper bound on the number of elements that it can host. To achieve that, the user will specify a `size <= 0` before building the cache.

### CODE REVIEW

File: simple.go [hunk 1]
```diff
@@ -18,7 +18,11 @@ func newSimpleCache(cb *CacheBuilder) *SimpleCache {
 }
 
 func (c *SimpleCache) init() {
-	c.items = make(map[interface{}]*simpleItem, c.size)
+	if c.size == -1 {
```
Comment (bluele) [inline simple.go]: I want `size==0` to means unbounded, so could you change to `c.size <= 0` ?

File: simple.go [hunk 2]
```diff
@@ -58,7 +62,7 @@ func (c *SimpleCache) set(key, value interface{}) (interface{}, error) {
 		item.value = value
 	} else {
 		// Verify size not exceeded
-		if len(c.items) >= c.size {
+		if (len(c.items) >= c.size) && c.size != -1 {
```
Comment (bluele) [inline simple.go]: @aaronwinter 
ref: https://github.com/bluele/gcache/pull/38/files#r127405430

Could you change to `if c.size > 0 && len(c.items) >= c.size` ?

File: simple.go (hunk 1 above)
Comment (aaronwinter) [inline simple.go]: 👍

Comment (kogan69) [inline simple.go]: Hi,
When do you guys expect this to be merged?
This is exactly the change I've been waiting for...

Comment (aaronwinter) [reply simple.go]: @kogan69 Hi, my bad sort of forgot about it. I will complete the PR by Wednesday this week.

**Comment by kogan69** [comment]
Thank you so much!


Cheers,
LK

**Comment by aaronwinter** [comment]
@kogan69 I fixed the PR. Let me add a test suite for it and then it should be good to go for review and merge.

**Comment by aaronwinter** [comment]
Added the test case, I'd like to add another to test for `SetWithExpire` but turns out the fake clock object we introduced is only package-exposed. I'll do another PR once we get that figured out.

This one is ready for review @bluele

**Comment by bluele** [comment]
Thanks, @aaronwinter !
LGTM.

**Pull request merged**